kube/minikube/scripts/deploy.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+# Function to check if a resource exists
+check_resource_exist() {
+  local kind=$1
+  local name=$2
+
+  # Wait for the resource to be available
+  while [[ -z "$(kubectl get $kind $name -o json)" ]]; do
+    echo "Waiting for $kind $name to be available..."
+    sleep 5
+  done
+
+  echo "$kind $name is available!"
+}
+
+# Function to check if a resource is ready (for Deployments)
+check_deployment_ready() {
+  local name=$1
+
+  # Wait for the Deployment to have available replicas
+  while [[ "$(kubectl get deployment $name -o 'jsonpath={.status.availableReplicas}')" != "$(kubectl get deployment $name -o 'jsonpath={.status.replicas}')" ]]; do
+    echo "Waiting for Deployment $name to have available replicas..."
+    sleep 5
+  done
+
+  echo "Deployment $name is ready!"
+}
+
+# Function to check if a resource is completed (for Jobs)
+check_job_completed() {
+  local name=$1
+
+  # Wait for the Job to be completed
+  while [[ "$(kubectl get job $name -o 'jsonpath={.status.conditions[?(@.type=="Complete")].status}')" != "True" ]]; do
+    echo "Waiting for Job $name to be completed..."
+    sleep 5
+  done
+
+  echo "Job $name is completed!"
+}
+
+# Function to apply a resource and check if it is ready or completed (for Jobs)
+apply_and_check_resource() {
+  local file=$1
+
+  kubectl apply -f "$file"
+
+  # Get the resource kind and name from the file
+  kind=$(kubectl get -f "$file" -o 'jsonpath={.kind}')
+  name=$(kubectl get -f "$file" -o 'jsonpath={.metadata.name}')
+
+  case "$kind" in
+    "Job")
+      check_job_completed "$name"
+      ;;
+    "Deployment")
+      check_deployment_ready "$name"
+      ;;
+    *)
+      ;;
+  esac
+}
+
+# Function to check if a secret 'kci-api-token' exists
+check_secret_key_exists() {
+  local kind=$1
+  local name=$2
+
+  # Wait for the resource to be available
+  while [[ -z "$(kubectl get $kind $name -o jsonpath='{.data.*}')" ]]; do
+    echo "Waiting for $kind $name to be available..."
+    sleep 5
+  done
+
+  echo "$kind $name is available!"
+}
+
+# Fork GitHub repository in Minikube node
+kubectl apply -f ../init/init-pod.yaml
+check_job_completed "github-cloning-pod"
+
+# Create Kubernetes secret
+kubectl create secret generic kernelci-pipeline-secrets --from-literal=kci-api-token=$KCI_API_TOKEN
+check_secret_key_exist secret kernelci-pipeline-secrets -o jsonpath='{.data.*}'
```

Comment:
Just got k3s testing now and got following while running script:
```
# ./deploy.sh
job.batch/pipeline-github-cloning-job created
job.batch/pipeline-github-cloning-job condition met
Init job is completed successfully
secret/kernelci-pipeline-secrets created
./deploy.sh: line 45: check_secret_key_exist: command not found
configmap/kernelci-pipeline-config created
configmaps kernelci-pipeline-config is available!
deployment.apps/kernelci-pipeline-monitor created
deployment.apps/kernelci-pipeline-monitor condition met
deployment.apps/kernelci-pipeline-regression-tracker created
deployment.apps/kernelci-pipeline-regression-tracker condition met
deployment.apps/kernelci-pipeline-scheduler created
deployment.apps/kernelci-pipeline-scheduler condition met
deployment.apps/kernelci-pipeline-scheduler configured
deployment.apps/kernelci-pipeline-scheduler condition met
deployment.apps/kernelci-pipeline-scheduler configured
deployment.apps/kernelci-pipeline-scheduler condition met
deployment.apps/kernelci-pipeline-tarball created
deployment.apps/kernelci-pipeline-tarball condition met
deployment.apps/kernelci-pipeline-test-report created
deployment.apps/kernelci-pipeline-test-report condition met
deployment.apps/kernelci-pipeline-timeout-closing created
deployment.apps/kernelci-pipeline-timeout-closing condition met
deployment.apps/kernelci-pipeline-timeout created
deployment.apps/kernelci-pipeline-timeout condition met
deployment.apps/kernelci-pipeline-timeout-holdoff created
deployment.apps/kernelci-pipeline-timeout-holdoff condition met
deployment.apps/kernelci-pipeline-trigger created
deployment.apps/kernelci-pipeline-trigger condition met
All components deployed successfully!
```
I think there is small typo in function name, s is missing

Reply:
So if this deployment works as-is both in MiniKube and k3s then it's already vendor-agnostic.  Is that right?

Reply:
(Well pending it actually works as there's a problem in the script apparently)

Reply:
Thanks for the review @nuclearcat 
Fixed it.

Reply:
Yes, correct @gctucker , it works with k3s as well.